### PR TITLE
Relax Verify Thicket Structure Check for Stats Functions

### PR DESCRIPTION
**Pull request by Treece-Burgess**
This PR contains a single change which will be outlined below. 

1. Update to `verify_thicket_structure` check within stats functions
    - Removed `profile` from the list of indexes to check for, this allows users to join on any metadata column they would like now.

### CODE REVIEW

File: thicket/stats/calc_boxplot_statistics.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from ..utils import verify_thicket_structures
 
 
-def calc_boxplot_statistics(thicket, columns=[], quartiles=[0.25, 0.5, 0.75], **kwargs):
```
Comment (michaelmckinsey1) [inline thicket/stats/calc_boxplot_statistics.py]: Should this be a separate PR?

Comment (Treece-Burgess) [reply thicket/stats/calc_boxplot_statistics.py]: Originally this PR was for both the `**kwargs` and for relaxing the verify thicket structure checks. However, I changed it to just be for relaxing the verify thicket structure check. Line 12 that you are pointing out, was apart of an old push, but I just double checked the current files and I do not see the `**kwargs` within the list of arguments.

Comment (michaelmckinsey1) [reply thicket/stats/calc_boxplot_statistics.py]: [Develop](https://github.com/LLNL/thicket/blob/develop/thicket/stats/calc_boxplot_statistics.py#L12) shows it in the current code and that's why it's in the diff. I think it's reasonable to have it in here if the title/description is updated or if it is separated out into a different PR; whichever is easier.

Comment (Treece-Burgess) [reply thicket/stats/calc_boxplot_statistics.py]: I see what you mean now, I will wait to remove that with another PR when the **kwargs are addressed. Thanks for pointing that out to me.

**Review by michaelmckinsey1 — APPROVED** [review]

**Review by slabasan — APPROVED** [review]
